.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+stages:
+  - test
+  - deploy
+
+test:
+  stage: test
+  image: python:3.8.7-slim-buster
+  before_script:
+    - python -m pip install --upgrade pip
+    - pip install pytest certifi
+    - if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - python setup.py install
+    # reduce the default SSL security level to get around a misconfiguration in the Ensembl server
+    # https://github.com/Ensembl/ensembl-rest/issues/427
+    # See https://askubuntu.com/a/1233456
+    - sed -i 's/DEFAULT@SECLEVEL=2/DEFAULT@SECLEVEL=1/' /usr/lib/ssl/openssl.cnf
+  script:
+    - PYTHONPATH=. pytest tests
+  environment:
+    name: test-env
+  only:
+    - external_pull_requests
+    - tags
+    - master
+
+
+# Not executed, parent job definition
+.deploy-python-tool:
+  stage: deploy
+  image: python:3.8.7-slim-buster
+  environment:
+    name: $ENVIRONMENT_NAME
+  script:
+    # Start ssh-agent and set identity then copy the Jar file and create a symlink
+    # See https://docs.gitlab.com/ee/ci/ssh_keys/
+    - apt-get update
+    - apt-get -y install openssh-client
+    - eval "$(ssh-agent)"
+    - echo "$EVA_ETL_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ${CLUSTER_USER}@${CLUSTER_LOGIN_NODE} "${EVA_SUBMISSION_PATH}/deploy.sh ${DEPLOY_POINT}"
```

Comment:
I don't really like relying on the `deploy.sh` script in the cluster that is also not into our version control. Maybe in a subsequent ticket it could be changed.

Maybe having the script in the `eva_submission` repo and copying to the cluster when deploying so we can keep track of it and use the latest version. Another approach would be to create tasks that are stored in the repo with all the deployment logic and we just call those from gitlab (This approach allow us to use those tasks by themselves and not completely rely con gitlab if we ever want to move from it)